nixos/tests/common/acme/client/default.nix

```diff
@@ -9,6 +9,11 @@ in {
     defaults = {
       server = "https://${caDomain}/dir";
       email = "hostmaster@example.test";
+      # Avoid a random 0-8 minute sleep when testing renewals.
+      # We are not using LE servers in testing so this is not
+      # going to impact their load.
+      # See https://github.com/go-acme/lego/issues/1656
+      extraLegoRenewFlags = ["-no-random-sleep"];
```

Comment:
Since we set `RandomizedDelaySec=24h` It might be best to pass this by default. This way we aren't starting the CLI and holding resources open for a delay. Plus if people manually remove `RandomizedDelaySec` they probably don't want the delay anyways.

Side note: We should probably also set `FixedRandomDelay=true` so that we are renewing on a consistent interval rather than just averaging the right interval. For daily renewals it probably doesn't matter much but if you are doing weekly and get unlucky you may be waiting nearly 2 weeks for the next check.

Reply:
But `RandomizedDelaySec` is only 24 hours and doesn't delay the next time the timer is triggered, so the worst case would be `renewInterval` + 24 hours in-between checks (e.g. 8 days for `weekly`)? `FixedRandomDelay` shouldn't change the (average) time between triggers because they're set via `OnCalendar`

Reply:
Ah right. It can add at most 24h between triggers (last one rolls a 0h and the next one rolls a 23h59min). But I think it still makes sense to do `FixedRandomDelay` so that the time between is fairly consistent.

Reply:
Yeah, agreed on adding no-random-sleep to the default options. To avoid triggering renewals on all upgrading systems, I am hard-coding it into the module's `renewFlags`. A random sleep during the execution of lego will needlessly delay the runtime of the service thus also delaying startup times for dependent systemd services, which is definitely unwanted in all cases.

I'll add FixedRandomDelay for the sake of it too.